### PR TITLE
Fix weather command, add ratelimit handling for POSTs

### PR DESCRIPTION
**Pull request by lbatalha**
This adds:
- weather command now uses textual output since upstream `png` output is disabled
- POST handling now deals with ratelimit bucket headers
- Added WET and WEST timezones to time command

### CODE REVIEW

File: utils.py [hunk 1]
```diff
@@ -157,27 +159,25 @@ def time(cmd):
...
-		flags = split[-1][1:]
-		location = ' '.join(split[:-1])
-	else:
-		flags = '1Fp'
-		location = cmd.args
-		if location.isdecimal() and len(location) == 5:
-			location += '-us'
-	url = 'https://wttr.in/%s.png?%s' % (urllib.parse.quote_plus(location), flags)
+	flags = 'format=**%l:**+%c+++🌡+`%t(%f)`++💦+`%h`++💨+`%w`++**☔**+`%p/3h`++**UVI:**+`%u`\
+				\n**Time:**+`%T`++**Sunrise:**+`%S`++**Sunset:**+`%s`++**Moon:**+%m'
```
Comment (raylu) [inline utils.py:163]: I... wasn't aware this was allowed. and it just dedents the whitespace on the second line or something?
can we uh
```suggestion
	flags = 'format=**%l:**+%c+++🌡+`%t(%f)`++💦+`%h`++💨+`%w`++**☔**+`%p/3h`++**UVI:**+`%u`\n' \
			'**Time:**+`%T`++**Sunrise:**+`%S`++**Sunset:**+`%s`++**Moon:**+%m'
```

File: utils.py [hunk 2]
```diff
@@ -157,27 +159,25 @@ def time(cmd):
...
+		if response.status_code == 503:
+			cmd.reply(f'{cmd.sender["pretty_name"]}: service unavailable for {location}')
+		elif response.status_code == 404:
+			cmd.reply(f'{cmd.sender["pretty_name"]}: {location} not found')
+		else:
+			cmd.reply(f'{cmd.sender["pretty_name"]}: error getting weather at {url}',
+					{'description': f'```{traceback.format_exc()[-500:]}```'})
 		return
-	cmd.reply(None, files={'weather.png': response.content})
+	cmd.reply(response.content.decode())
```
Comment (raylu) [inline utils.py:180]: that's some dangerous system encoding you got there
```suggestion
	cmd.reply(response.text)
```

File: utils.py [hunk 3]
```diff
@@ -157,27 +159,25 @@ def time(cmd):
...
+	url = f'https://wttr.in/{urllib.parse.quote_plus(location)}?{flags}'
 	try:
 		response = rs.get(url)
 		response.raise_for_status()
 	except Exception:
-		cmd.reply('%s: error getting weather at %s' % (cmd.sender['pretty_name'], url),
-				{'description': '```%s```' % traceback.format_exc()[-500:]})
+		if response.status_code == 503:
+			cmd.reply(f'{cmd.sender["pretty_name"]}: service unavailable for {location}')
+		elif response.status_code == 404:
```
Comment (raylu) [inline utils.py:174]: this feels kinda questionable... but it does save repeating `return`s. the alternative is
```py
try:
	response = rs.get(url)
	if response.status_code == 503:
		cmd.reply(f'{cmd.sender["pretty_name"]}: service unavailable for {location}')
		return
	elif response.status_code == 404:
		cmd.reply(f'{cmd.sender["pretty_name"]}: {location} not found')
		return
    response.raise_for_status()
except Exception:
	cmd.reply(f'{cmd.sender["pretty_name"]}: error getting weather at {url}',
			{'description': f'```{traceback.format_exc()[-500:]}```'})
    return
```
I leave it up to you to decide which is better
(or, you could introduce some kinda `BotCommandException` and change how the bot handles exceptions... would clean up the control flow a bit)

File: utils.py (hunk 1 above)
Comment (lbatalha) [reply utils.py]: https://github.com/raylu/sbot/actions/runs/8520604984/job/23337068220?pr=23#step:6:16
Linter does not seem to like this

File: utils.py (hunk 3 above)
Comment (lbatalha) [reply utils.py]: Yours is better, I think, it avoids raising exceptions for things that are actually handled properly.

**Pull request merged**